VirtoCommerce.Platform.Data.Assets.FileSystem/FileSystemBlobProvider.cs

```diff
@@ -141,70 +147,77 @@ public virtual BlobSearchResult Search(string folderUrl, string keyword)
                     Url = GetAbsoluteUrlFromPath(file),
                     ContentType = MimeTypeResolver.ResolveContentType(fileInfo.Name),
                     Size = fileInfo.Length,
-                    FileName = fileInfo.Name,
+                    Name = fileInfo.Name,
                     ModifiedDate = fileInfo.LastWriteTimeUtc
                 };
                 blobInfo.RelativeUrl = GetRelativeUrl(blobInfo.Url);
-                retVal.Items.Add(blobInfo);
+                retVal.Results.Add(blobInfo);
             }
-            return retVal;
+
+            retVal.TotalCount = retVal.Results.Count();
+            return Task.FromResult(retVal);
         }
 
         /// <summary>
         /// Create folder in file system within to base directory
         /// </summary>
         /// <param name="folder"></param>
-        public virtual void CreateFolder(BlobFolder folder)
+        public virtual async Task CreateFolderAsync(BlobFolder folder)
         {
-            if (folder == null)
-            {
-                throw new ArgumentNullException("folder");
-            }
-            var path = _storagePath;
-            if (folder.ParentUrl != null)
+            await Task.Factory.StartNew(() =>
             {
-                path = GetStoragePathFromUrl(folder.ParentUrl);
-            }
-            path = Path.Combine(path, folder.Name);
-
-            ValidatePath(path);
+                if (folder == null)
+                {
+                    throw new ArgumentNullException("folder");
+                }
+                var path = _storagePath;
+                if (folder.ParentUrl != null)
+                {
+                    path = GetStoragePathFromUrl(folder.ParentUrl);
+                }
+                path = Path.Combine(path, folder.Name);
 
-            if (!Directory.Exists(path))
-            {
-                Directory.CreateDirectory(path);
-            }
+                ValidatePath(path);
 
+                if (!Directory.Exists(path))
+                {
+                    Directory.CreateDirectory(path);
+                }
+            });
         }
 
         /// <summary>
         /// Remove folders and blobs by absolute or relative urls
         /// </summary>
         /// <param name="urls"></param>
-        public virtual void Remove(string[] urls)
+        public virtual async Task RemoveAsync(string[] urls)
         {
-            if (urls == null)
-            {
-                throw new ArgumentNullException("urls");
-            }
-            foreach (var url in urls)
-            {
-                var path = GetStoragePathFromUrl(url);
-
-                ValidatePath(path);
+            await Task.Factory.StartNew(() => {
```

Comment:
Need to remove `async` modification and you will be able to use synchronous code inside this method and remove Task.Factory.StartNew expression